### PR TITLE
Ephedrine Changes

### PR DESCRIPTION
**Pull request by wesoda25**
:cl: wesoda25
balance: lowered ephedrine addiction and overdose amount
balance: deadlier ephedrine overdose
:cl:

Pentetic Acid OD removed. If that was your issue with this PR, I'd appreciate if you removed the downvote because the main point of this PR was ephedrine. 

And ephedrine is powergamer tool # 1. I'm currently trying to get Nuka Cola removed, and people in there are saying "I'll just use ephedrine". It has no medicinal purpose, and is relatively unbalanced. I believe it was mostly intended to be an ingredient in other chems. So I'm changing the addiction and OD levels to pretty low, and the OD hits hard as shit. It won't kill you fast, but its honestly easier to get cloned if you start ODing.

### CODE REVIEW

**Comment by tuypo1** [comment]
If it had no purpose they wouldn't give everybody a pen of it at round start.

**Comment by wesoda25** [comment]
> If it had no purpose they wouldn't give everybody a pen of it at round start.

fun fact ephedrine is not the same thing as epinephrine. so you know

**Comment by subject217** [comment]
eph overdose heart attack is pretty epic not gonna lie

jamming that into a shotgun dart would be funny

**Comment by necromanceranne** [comment]
The way I use ephedrine, in a cigarette, is in no way negatively affected by this change, so I can keep powergaming it in large quantities. What is basically budget meth remains as strong as ever, you just have to be more precise with mixes. 

Ephedrine metabolizes slower than other chems, so 14u lasts a lot longer than you think.

**Comment by subject217** [comment]
The best way to use Ephedrine has always been to suck 100u of it into a vape. This PR doesn't affect that either. That being said, cigarette reagent microdosing is a completely separate issue and really just a limitation of goofchem in it's current iteration. It would probably help somewhat if reagents had thresholds before they would act. Sort of like how addiction thresholds work, but just for the reagent doing anything. Could have multiple stages too. I don't think it'd be that complex code wise.

**Comment by necromanceranne** [comment]
But then you might inadvertently hurt legitimate uses of microdosing, like cycling slime types via slime mutation toxins. I believe cigarettes have code already for dilution but I'm not sure it's actually working because at the moment it consistently injects .2 u of any other chem it contains and .15 nicotine. .2 is the metabolizism rate of ephedrine so it works consistently.

**Comment by Shardj** [comment]
"will purge explosive death mixes before blowing one up"

I thought the explode on ingestion trick (cooled explosive meth) was removed a year and a half ago when your body temp no longer heats up chemicals inside you or something

**Comment by necromanceranne** [comment]
Black powder still ignites in the blood

**Comment by Shardj** [comment]
Only if you're on fire though from what I've seen, it takes quite a while to explode from that anyway. Also worthless if shot in a syringe, if used in a smoke (or foam) I guarantee pentic acid wont get it all out if the smoke is made properly and keeps you in it with lube. But I get what you were trying to say there.

Out of interest do you know the specific pull request which stopped explosive meth from blowing up on ingestion? I can't remember what they changed and I'm interested now, I used to like using it.

**Comment by DaxDupont** [comment]
Don't nerf penetic acid you incompetent medbay player.

There's only a few things that clear radiation and one of them is penetic acid, a lot of rads require about 60u+ of the shit if not more. Even with that and players running around with contamination makes things worse. Radiation is already shit to deal with because one chumbis can get the entire station radiated.

**Comment by DaxDupont** [comment]
What's with all the ided prs lately

**Comment by 81Denton** [comment]
>Also worthless if shot in a syringe

BP mixes in 15u syringes easily kill and delimb/husk if done right

**Comment by Shardj** [comment]
@81Denton what are you using in that mix? Sounds like fun, even when you could use meth explosion syringes for a full 15u it didn't do that

**Comment by Hathkar** [comment]
Like others have said, changing the OD/Addiction thresholds on ephedrine won't stop people from abusing it. Why not just nerf the stun resistance and movespeed bonuses?

**Comment by FloranOtten** [comment]
This just screams "MY ASSASINATION PLAN FAILED AND I'M SALTY"

Don't nerf because you're salty. Get a better plan for next time.

**Comment by AlexTheSysop42** [comment]
Maybe don't deathmix near a doctor?????

That said, pentetic is the _only_ thing that can cure large amounts of rads (afaik) - potassium iodide is way too weak

Speaking of which, pentetic acid in a syringe gun is a pretty good way to mess up anyone using ephedrine

**Comment by Teepeepee** [comment]
good luck landing it on sonic speed running people lmao

**Comment by kevinz000** [comment]
pentetic acid nerf is bad
ephedrine nerfs are always good

**Comment by Vidiveni** [comment]
I agree on the ephedrine nerf since it's just so powerful and it was destined to happen sooner or later. It's not even a major nerf. What wesoda is doing might save ephedrine from being overnerfed by someone else. 

But I think a bigger problem is the actual strength of the speed boosts.

**Comment by AlexTheSysop42** [comment]
Speed boosts are only strong because movespeed was lowered and they weren't scaled  at all

**Comment by 81Denton** [comment]
This doesn't help with cig/vape microdosing. 
I'm not 100% up to speed with how cigs/vapes process reagents, but adding a cycle based OD like morphine sleep might help.

**Comment by 81Denton** [comment]
Alternatively, you could always make ephedrine metabolize into a second drug that has its own OD.

**Comment by wesoda25** [comment]
> Only if you're on fire though from what I've seen, it takes quite a while to explode from that anyway. Also worthless if shot in a syringe, if used in a smoke (or foam) I guarantee pentic acid wont get it all out if the smoke is made properly and keeps you in it with lube. But I get what you were trying to say there.
> 
> Out of interest do you know the specific pull request which stopped explosive meth from blowing up on ingestion? I can't remember what they changed and I'm interested now, I used to like using it.
> 
> Edit: #29968 what i was looking for, mob_react was simply set to false for /datum/chemical_reaction/reagent_explosion/methsplosion

 I know a specific, round start recipe which fits ina syringe and blows up the person, killing them (doesn’t gib).

**Comment by wesoda25** [comment]
Anyways someone suggested I should nerf pentetic given how strong of a chem it is, and I went along with it. I still don’t think this poses that much of a problem, make 20u pills and just use them once the amount in your system is <5. However if its so strongly opposed I’ll remove that part of the PR. 

The ephedrine nerf forces the player to attempt to microdose. If not, they get these crazy OD effects which usually end in a heart attack. I have no idea how to go about liming microdosing without just lowering OD level to 0 or straight up nerfing the effects of ephedrine. I toyed with the idea of only making it work when there was more than a certain amount in your system, which could easily lead to mistakes and overdoses, but as to how elludes me. I can’t think of any examples in the code which work like this, and my code style is literally copy, paste, and edit.

**Comment by AdamElTablawy** [comment]
cryo syringes can carry up to 20u of chemicals for no material cost

essentially, this would allow a chemist to shoot pseudo initropidil at people via syringe gun pretty quickly. it wouldn't kill them immediately but it would hospitalize them.

also considering how long pentetic acid takes to purge, 4 toxin damage per tick is absolutely fucking insane. it's absolutely fucking insane for any chemical because even most poisons aren't that deadly, but this would basically annihilate anyone it hits, without any chance of being able to treat the person because it purges everything, including antitoxins

pentetic acid is also comparatively difficult to make, so it's not much of a problem for deathmixes

if your deathmix doesn't kill someone in less time than it takes to get them into a sleeper, it's a bad deathmix, and it needs an upgraded sleeper, so that's banking on medbay being competent, which is rare in and of itself

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -567,6 +573,27 @@
 	return TRUE
 
 /datum/reagent/medicine/ephedrine/overdose_process(mob/living/M)
+	if(prob(30) && iscarbon(M))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]:
```DM
if(iscarbon(M))
    if(your_first_prob)
        your_first_prob_code_here
    if(your_second_prob)
        your_second_prob_code_here
    if(your_third_prob)
        your_third_prob_code_here

if(the_other_prob)
    the_other_prob_code_here
```

**Comment by imsxz** [comment]
better solution would be to make addictions/overdose be counted towards by processing chemicals rather than just having them

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: just fyi this code assumes you wanted all of these to possibly trigger at once.

**Comment by wesoda25** [comment]
> cryo syringes can carry up to 20u of chemicals for no material cost
> 
> essentially, this would allow a chemist to shoot pseudo initropidil at people via syringe gun pretty quickly. it wouldn't kill them immediately but it would hospitalize them.
> 
> also considering how long pentetic acid takes to purge, 4 toxin damage per tick is absolutely fucking insane. it's absolutely fucking insane for any chemical because even most poisons aren't that deadly, but this would basically annihilate anyone it hits, without any chance of being able to treat the person because it purges everything, including antitoxins
> 
> pentetic acid is also comparatively difficult to make, so it's not much of a problem for deathmixes
> 
> if your deathmix doesn't kill someone in less time than it takes to get them into a sleeper, it's a bad deathmix, and it needs an upgraded sleeper, so that's banking on medbay being competent, which is rare in and of itself

Pentetic isn’t a wall block, meaning you can spam the person with charcoal, which has no OD attached to it. I purposefully intended for it to be a very deadly overdose. Again, I’ll probably subtract it from the PR. 

Also pentetic is braindead easy to make. 2 minute recipe. 

Pentetic OD threshold is 25, so I assume you are referring to ephedrine for a cryo syringe. Its not a plausible counter argument, as you said, any good deathmix kills in under 10 seconds. Ephedrine OD fucks with someone hard, but the only way to die from it is about 5 minutes in, when heart failure hits.

**Comment by AdamElTablawy** [comment]
yeah, but it's still an easy way to induce heart attacks. releasing this shit in a smoke grenade or via syringe spam is virtually going to guarantee medbay gets flooded with people having heart attacks five minutes later before they realize there's any problem and is kind of disgusting all around

like yeah it's not instantly lethal but it's a really annoying grief chem that people can use to cause mass chaos, like skewium but way easier to make 

and someone actually abusing it could skimp on the heart attack issue by exercising unless i'm misreading something

either way, might be a good idea to change it from 'heart attack' to stamina damage or brain damage or some such

**Comment by Vidiveni** [comment]
The primary use for pentetic acid for me is to fight radiation, and that requires huge doses of pentetic acid. 

Remove the purge effect from pentetic acid instead imo. Makes calomel useful. Makes pentetic useful in a heal-mix. 

Then double the metabolism of pentetic to 0.4 so it doesn't become OP antitoxin.

**Comment by wesoda25** [comment]
> yeah, but it's still an easy way to induce heart attacks. releasing this shit in a smoke grenade or via syringe spam is virtually going to guarantee medbay gets flooded with people having heart attacks five minutes later before they realize there's any problem and is kind of disgusting all around
> 
> like yeah it's not instantly lethal but it's a really annoying grief chem that people can use to cause mass chaos, like skewium but way easier to make
> 
> and someone actually abusing it could skimp on the heart attack issue by exercising unless i'm misreading something
> 
> either way, might be a good idea to change it from 'heart attack' to stamina damage or brain damage or some such

I disagree. It makes sense that a strong stimulant could cause a heart attack. To be honest, chance of heart attack is too high, I might lower it. You can literally do gibsmoke putting this in a smoke grenade isnt that OP. 

I think exercising is a preventative measure, not cure, so in theory yes, someone could work out before taking ephedrine. However, the seizures and oxy loss should fuck them over hard.

**Comment by Kierany9** [comment]
The only way to fix antistun chemicals is to purge them all

I spent a hell of a long time trying the other way, it does not work. Powergamers will always find a loophole or a way to abuse it while rendering it so finicky to use it saps all the fun out of it, just nuke it now and get it over with.

-t ex hippie coder

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (Shardj) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: He means the repeated iscarbon checks from line 570 to 585 are uneeded

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: You can avoid going through the probability check only to fail the carbon check x3 if you put the carbon check at the front. I'm fully aware both "work".

**Comment by subject217** [comment]
the nuka cola change will need a changelog

**Comment by subject217** [comment]
yuh yikies, should close it on principle so he learns why you don't do it

**Comment by YakumoChen** [comment]
>stealth removal of nuka cola

**Comment by wesoda25** [comment]
Fucking dammit just forget this shit I’ll be back

**Pull request closed without merging**